Makefile

```diff
@@ -56,33 +57,42 @@ test-generate: generate
 test-unit:
 	@go test -short ./...
 
-test-system: build bin/minio
+test-system: builddev bin/minio
 	@ZTEST_PATH="$(CURDIR)/dist:$(CURDIR)/bin" go test .
 
-test-run: build bin/minio
+test-run: builddev bin/minio
 	@ZTEST_PATH="$(CURDIR)/dist:$(CURDIR)/bin" go test . -run $(TEST)
 
-test-heavy: build
+test-heavy: builddev
 	@PATH="$(CURDIR)/dist:$(PATH)" go test -tags=heavy ./mdtest
 
 .PHONY: test-services
-test-services: build
+test-services: builddev
 	@ZTEST_PATH="$(CURDIR)/dist:$(CURDIR)/bin" \
 		ZTEST_TAG=services \
 		go test -run TestZed/ppl/zqd/ztests/redis .
 
-perf-compare: build $(SAMPLEDATA)
+perf-compare: builddev $(SAMPLEDATA)
 	scripts/perf-compare.sh
 
-output-check: build $(SAMPLEDATA)
+output-check: builddev $(SAMPLEDATA)
 	scripts/output-check.sh
 
 # If the build recipe changes, please also change npm/build.
 build: $(PEG_DEP)
+	@mkdir -p dist
+	@go build -ldflags='$(LDFLAGS)' -o dist $(BUILD_COMMANDS)
+
+.PHONY: builddev
+builddev: $(PEG_DEP)
```

Comment:
Do we really need both `build` and `builddev`? If not, I'd much prefer to leave `build` as is.

Reply:
Sure